lab-raziyeh/test/book-test.js

```diff
@@ -0,0 +1,169 @@
+'use strict';
+
+const request = require('superagent');
+const expect = require('chai').expect;
+const Book = require('../model/book.js');
+const url = 'http://localhost:3000';
+const exampleBook = {
+  title: 'JavaScript-2',
+  author: 'Raziyeh Bazargan',
+  page: '120',
+  creationDate: '2016-09-26T23:39:59.775Z',
+};
+
+//start the server
+require('../server.js');
+
+describe('testing Book Routes', function(){
+  var tempBook;
+  describe('GET requests tests', function(){
+    describe('testing GET /api/book/unregistered', function() {
+      it('expecting to return an error with unregistered route - GET', function(done){
+        request.get(`${url}/api/woow`)
+        .end((err, res) => {
+          expect(res.status).to.equal(404);
+          expect(res.err).to.not.null;
+          done();
+        });
+      });
+    });
+
+    describe('testing GET /api/book valid id provided in the request', function(){
+      before(done => {
+        Book.createItem(exampleBook)
+          .then( book => {
+            tempBook = book;
+            console.log('***********',book);
```

Comment:
@Raziyehbazargan good use of the asterisks to see where you are at in your terminal - manual debugging at it's finest!